hack/charts/hyperparameter-tuning-jobs/templates/hpo-job.yaml

```diff
@@ -3,72 +3,76 @@ kind: HyperparameterTuningJob
-         {{- end }}
-      algorithmSpecification:
-        trainingImage: {{ .Values.spec.image }}
-        trainingInputMode: {{ .Values.spec.traininginputmode }}
-      roleArn: {{ .Values.spec.rolearn }}
-      inputDataConfig:
-            {{- range .Values.spec.inputdataconfig }}
-              - channelName: {{ .channel }}  
-                dataSource: 
-                  {{- if .s3 }}
-                  s3DataSource:
-                     s3DataType: {{ .s3.s3datatype | default "S3Prefix" }}
-                     s3Uri: {{ .s3.s3uri }}
-                     s3DataDistributionType: {{ .s3.s3datadistributiontype | default "FullyReplicated" }}
-                  {{- end }}
-                  {{- if .filesystem }}
-                  fileSystemDataSource: 
-                      fileSystemId: {{ .filesystem.filesystemid }}
-                      fileSystemAccessMode: {{ .filesystemaccessmode | default "ro" }}
-                      fileSystemType: {{ .filesystem.filesystemtype }}
-                      directoryPath: {{ .filesystem.directorypath }}
-                  {{- end }}
-                contentType: {{ .contenttype | default "text/csv" }}
-                compressionType: {{ .compressiontype | default "None" }}
-                recordWrapperType: {{ .recordwrappertype | default "None" }}
-                inputMode: {{ .inputmode | default "File" }}
+  hyperParameterTuningJobConfig:
+    resourceLimits:
+      maxNumberOfTrainingJobs:  {{ .Values.spec.hyperparameterTuningJobConfig.maxNumberOfTrainingJobs}}
+      maxParallelTrainingJobs:  {{ .Values.spec.hyperparameterTuningJobConfig.maxParallelTrainingJobs }}
+    strategy: {{ .Values.spec.hyperparameterTuningJobConfig.strategy | quote }}
+  region: {{ .Values.spec.region }}
+  hyperParameterTuningJobConfig:
+    strategy: {{ .Values.spec.hyperparameterTuningJobConfig.strategy | quote }}
+    hyperParameterTuningJobObjective:
+      type: {{ .Values.spec.hyperparameterTuningJobConfig.hyperparameterTuningJobObjective.type }}
+      metricName:  {{ .Values.spec.hyperparameterTuningJobConfig.hyperparameterTuningJobObjective.metricName }}
+    resourceLimits:
+      maxNumberOfTrainingJobs: {{ .Values.spec.hyperparameterTuningJobConfig.maxNumberOfTrainingJobs}}
+      maxParallelTrainingJobs: {{ .Values.spec.hyperparameterTuningJobConfig.maxParallelTrainingJobs }}
+    parameterRanges:
+      integerParameterRanges:
+        - name:  {{ .Values.spec.hyperparameterTuningJobConfig.integerParameterRanges.name }}
+          minValue: {{ .Values.spec.hyperparameterTuningJobConfig.integerParameterRanges.minValue | quote }}
+          maxValue: {{ .Values.spec.hyperparameterTuningJobConfig.integerParameterRanges.maxValue | quote }}
+          scalingType: {{ .Values.spec.hyperparameterTuningJobConfig.integerParameterRanges.scalingType }}
+      # Add support for next two ranges in values.yaml
+      continuousParameterRanges: []
+      categoricalParameterRanges: []
```

Comment:
This is todo ? @cadedaniel  can u confirm that spec will work just that helm chart needs policing.

Reply:
Yeah nick made a task for these https://issues.amazon.com/issues/P32170625